libraries/Microsoft.Bot.Configuration/BotConfiguration.cs

```diff
@@ -110,20 +121,41 @@ public static async Task<BotConfiguration> LoadAsync(string file, string secret
         }
 
         /// <summary>
-        /// Save the file with secret
+        /// Load the configuration from a .bot file.
         /// </summary>
-        /// <param name="secret">secret for encryption</param>
+        /// <param name="file">Path to bot file. </param>
+        /// <param name="secret">Secret to use to decrypt the file on disk. </param>
+        /// <returns>BotConfiguration. </returns>
+        public static BotConfiguration Load(string file, string secret = null)
+        {
+            return BotConfiguration.LoadAsync(file, secret).GetAwaiter().GetResult();
+        }
+
+        /// <summary>
+        /// Save the file with secret.
+        /// </summary>
+        /// <param name="secret">Secret for encryption. </param>
         /// <returns>task</returns>
         public Task SaveAsync(string secret = null)
         {
             return this.SaveAsAsync(this.location, secret);
         }
 
+        /// <summary>
+        /// Save the file with secret.
+        /// </summary>
+        /// <param name="secret">Secret for encryption. </param>
+        public void Save(string secret = null)
+        {
+            this.SaveAsync(secret).GetAwaiter().GetResult();
+        }
+
         /// <summary>
         /// Save the configuration to a .bot file.
         /// </summary>
-        /// <param name="path">path to bot file</param>
-        /// <param name="secret">secret for encrypting the file keys.</param>
+        /// <param name="path">Path to bot file.</param>
+        /// <param name="secret">Secret for encrypting the file keys.</param>
+        /// <returns>Task. </returns>
         public async Task SaveAsAsync(string path, string secret = null)
         {
```

Comment:
Looking at the code below, "path" can be null as a valid parameter (and this.location is then used). If this is the case (and I don't like the pattern, but suspect we shouldn't change it at this time) then please make path an optional argument that defaults to null.

Also add in an argument check that verifies the overall state is valid (this.Location is set, if path is null).

Reply:
Added the state check. Same, I don't like the pattern. In this PR I'll do as many non-breaking improvements as I can. Then in a separate PR I might propose a better overall implementation for BotConfigation.